evalml/pipelines/components/estimators/classifiers/kneighbors_classifier.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from sklearn.neighbors import KNeighborsClassifier as SKKNeighborsClassifier
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+
+
+class KNeighborsClassifier(Estimator):
+    """
+    K-Nearest Neighbors Classifier.
+    """
+    name = "K-Nearest Neighbors Classifier"
```

Comment:
nit-pick: Can we change the name to simply be `KNN Classifier`? Thinking from the user perspective, it might be annoying to type `K-Nearest Neighbors Classifier` if they're defining a component graph or parameters, especially since none of our other components has a dash in it

Reply:
Can do!